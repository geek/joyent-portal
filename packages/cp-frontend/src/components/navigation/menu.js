import React from 'react';
import PropTypes from 'prop-types';
import forceArray from 'force-array';

import { LayoutContainer } from '@components/layout';

import {
  SectionList,
  SectionListItem,
  SectionListNavLink
} from 'joyent-ui-toolkit';

const getMenuItems = (...links) =>
  forceArray(links).map(({ pathname, name }) =>
    <SectionListItem key={pathname}>
      <SectionListNavLink activeClassName="active" to={pathname}>
        {name}
      </SectionListNavLink>
    </SectionListItem>
  );

const Menu = ({ links = [] }) =>
  <LayoutContainer plain>
    <SectionList>
      {getMenuItems(...links)}
    </SectionList>
  </LayoutContainer>;

Menu.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      pathname: PropTypes.string
    })
  )
};

export default Menu;
