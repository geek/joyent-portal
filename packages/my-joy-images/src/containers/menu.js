import React from 'react';
import get from 'lodash.get';

import Menu from '@components/menu';

const SECTIONS = [
  { name: 'Summary', pathname: 'summary' },
  { name: 'Tags', pathname: 'tags' }
];

export default ({ match }) => {
  const imageSlug = get(match, 'params.image');
  const sections = imageSlug !== '~create' ? SECTIONS : [];

  const links = sections.map(({ name, pathname }) => ({
    name,
    pathname: `/images/${imageSlug}/${pathname}`
  }));

  return <Menu links={links} />;
};
