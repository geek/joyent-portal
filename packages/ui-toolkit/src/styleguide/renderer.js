import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import remcalc from 'remcalc';
import is from 'styled-is';

import { Grid, Row, Col } from '../grid';
import theme from '../theme';

const Main = styled(Row)`
  padding-top: ${remcalc(40)};
`;

const Sticky = styled.div`
  position: sticky;
  top: 0;
  z-index: 9;

  ${is('sidebar')`
    top: ${remcalc(100)};
    max-height: 85vh;
    overflow: auto;
  `};
`;

const fullTheme = {
  ...theme,
  spacing: {
    0.5: remcalc(4),
    0: remcalc(0),
    1: remcalc(6),
    2: remcalc(12),
    3: remcalc(18),
    4: remcalc(24),
    5: remcalc(30),
    6: remcalc(36),
    7: remcalc(42),
    8: remcalc(48),
    9: remcalc(54),
    10: remcalc(60)
  }
};

const StyleGuideRenderer = ({
  title,
  homepageUrl,
  children,
  toc,
  hasSidebar
}) => (
  <ThemeProvider theme={fullTheme}>
    <Grid>
      <Main>
        {hasSidebar && (
          <Col xs={3}>
            <Sticky sidebar>{toc}</Sticky>
          </Col>
        )}
        <Col xs={hasSidebar ? 9 : 12} lg={hasSidebar ? 8 : 12}>
          {children}
        </Col>
      </Main>
    </Grid>
  </ThemeProvider>
);

export default StyleGuideRenderer;
