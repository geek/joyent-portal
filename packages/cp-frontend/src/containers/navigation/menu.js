import React from 'react';
import { connect } from 'react-redux';
import { compose } from 'react-apollo';
import withNotFound from './not-found-hoc';
import { Menu as MenuComponent } from '@components/navigation';

export const Menu = ({ location, match, sections }) => {

  if (!sections || !sections.length) {
    return null;
  }

  const sectionsWithPathnames = sections.map(section => {
    return {
      name: section.name,
      pathname: `${match.url}/${section.pathname}`
    };
  });
  return <MenuComponent links={sectionsWithPathnames} />;
};

const connectMenu = connect(
  (state, ownProps) => {
    const params = ownProps.match.params;
    const deploymentGroupSlug = params.deploymentGroup;
    const serviceSlug = params.service;

    if ((deploymentGroupSlug || '').match(/^~/)) {
      return {};
    }

    const sections = serviceSlug
      ? state.ui.sections.services
      : deploymentGroupSlug ? state.ui.sections.deploymentGroups : null;

    return {
      sections
    };
  },
  dispatch => ({})
);

export default compose(
  connectMenu,
  withNotFound()
)(Menu);
