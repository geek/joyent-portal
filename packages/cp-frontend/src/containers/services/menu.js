import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'react-apollo';
import { Col, Row } from 'react-styled-flexboxgrid';
import remcalc from 'remcalc';
import unitcalc from 'unitcalc';
import { LayoutContainer } from '@components/layout';
import { Title } from '@components/navigation';
import { withNotFound } from '@containers/navigation';

import { FormGroup, Toggle, ToggleList, Legend } from 'joyent-ui-toolkit';

const StyledLegend = Legend.extend`
  float: left;
  padding-top: ${unitcalc(2)};
  margin-right: ${unitcalc(1.5)};
`;

const PaddedRow = Row.extend`
  margin-bottom: ${remcalc(18)}
`;

export const ServicesMenu = ({
  location: { pathname },
  history: { push }
}) => {

  const toggleValue = pathname.split('-').pop();

  const handleToggle = evt => {
    const value = evt.target.value;
    if (value !== toggleValue) {
      const index = pathname.lastIndexOf('-');
      const path = `${pathname.slice(0, index)}-${value}`;
      push(path);
    }
  };

  return (
    <LayoutContainer plain>
      <Title>Services</Title>
      <PaddedRow>
        <Col xs={5}>
          <FormGroup name="service-view" value={toggleValue}>
            <StyledLegend>View</StyledLegend>
            <ToggleList>
              <Toggle value="list" onChange={handleToggle}>
                List
              </Toggle>
              <Toggle value="topology" onChange={handleToggle}>
                Topology
              </Toggle>
            </ToggleList>
          </FormGroup>
        </Col>
      </PaddedRow>
    </LayoutContainer>
  );
};

ServicesMenu.propTypes = {
  location: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
};

export default compose(withNotFound())(ServicesMenu);
