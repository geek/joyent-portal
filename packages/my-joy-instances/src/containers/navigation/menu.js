import { connect } from 'react-redux';
import get from 'lodash.get';

import { Menu } from '@components/navigation';

export default connect((state, { match }) => {
  const instanceSlug = get(match, 'params.instance');
  const allSections = get(state, 'ui.sections');
  const sections = instanceSlug ? allSections.instances : [];

  const links = sections.map(({ name, pathname }) => ({
    name,
    pathname: `/${instanceSlug}/${pathname}`
  }));

  return {
    links
  };
})(Menu);
