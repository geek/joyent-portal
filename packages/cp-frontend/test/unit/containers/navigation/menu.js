
/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Menu } from '@containers/navigation/menu.js';
import { Router, Store } from '../../mocks';

it('renders <Menu /> without throwing', () => {
  const tree = renderer
    .create(
      <Store>
        <Router>
          <Menu />
        </Router>
      </Store>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
