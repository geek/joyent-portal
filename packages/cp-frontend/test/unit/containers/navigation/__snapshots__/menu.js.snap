// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Menu /> without throwing 1`] = `null`;
